server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymsAction.java

```diff
@@ -103,14 +104,14 @@ public int hashCode() {
 
     public static class Response extends ActionResponse implements StatusToXContentObject {
 
-        private final Result result;
+        private final SynonymsManagementAPIService.UpdateSynonymsResult result;
```

Comment:
I did some refactoring for PUT. It didn't feel right that the `SynonymsManagementAPIService` depended on the request, so I defined the response enum in the `SynonymsManagementAPIService` and the response now uses it.